substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -22,56 +22,246 @@ use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
 
 #[cfg(feature = "try-runtime")]
 use sp_runtime::TryRuntimeError;
+pub mod v6 {
```

Comment:
I apologise for the confusing diff. All I did is to move the migration mods in order of recency (v6 then v5 ... v1) which seems much more readable and that's how we do it in other pallets. If you don't agree and like the older way, I am happy to revert the order and move v6 to the end of the file.